server/src/main/java/org/opensearch/action/search/SearchPhaseController.java

```diff
@@ -584,12 +584,14 @@ private static void validateMergeSortValueFormats(Collection<? extends SearchPha
                 firstResult = false;
                 ulFormats = new boolean[formats.length];
                 for (int i = 0; i < formats.length; i++) {
-                    ulFormats[i] = formats[i] == DocValueFormat.UNSIGNED_LONG_SHIFTED ? true : false;
+                    ulFormats[i] = (formats[i] == DocValueFormat.UNSIGNED_LONG_SHIFTED || formats[i] == DocValueFormat.UNSIGNED_LONG)
+                        ? true
+                        : false;
                 }
             } else {
                 for (int i = 0; i < formats.length; i++) {
                     // if the format is unsigned_long in one shard, and something different in another shard
-                    if (ulFormats[i] ^ (formats[i] == DocValueFormat.UNSIGNED_LONG_SHIFTED)) {
+                    if (ulFormats[i] ^ (formats[i] == DocValueFormat.UNSIGNED_LONG_SHIFTED || formats[i] == DocValueFormat.UNSIGNED_LONG)) {
```

Comment:
I think we can support this now with ```SortedWiderNumericSortField```.  
However, I think since in IndexNumericFieldData we specified ```SortField.Type```  as LONG, it will return data in LONG. Are we converting that LONG to unsigned LONG somewhere or if not, will there be precision loss ?

Reply:
> Are we converting that LONG to unsigned LONG somewhere or if not, will there be precision loss ?

Not for long conversions but for double conversions there would be precision loss (documented)